src/gui/pedalboard/cables/CableView.cpp

```diff
@@ -31,45 +31,46 @@ void CableView::paint (Graphics& g)
         }
     }
 
-    g.setColour (cableColour.brighter (0.1f));
-    for (auto* cable : cables)
+    if (cableMouse != nullptr)
     {
-        auto* startEditor = board->findEditorForProcessor (cable->startProc);
-        jassert (startEditor != nullptr);
-
-        auto startPortLocation = CableViewPortLocationHelper::getPortLocation ({ startEditor, cable->startIdx, false });
-
-        if (cable->endProc != nullptr)
+        auto mousePos = cableMouse->getPosition();
+        const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cables.getLast()->startProc);
+        if (nearestInputPort.editor != nullptr && nearestInputPort.isInput && ! portLocationHelper->isInputPortConnected (nearestInputPort))
         {
-            auto* endEditor = board->findEditorForProcessor (cable->endProc);
-            jassert (endEditor != nullptr);
+            auto endPortLocation = CableViewPortLocationHelper::getPortLocation (nearestInputPort);
+            drawCablePortGlow (g, endPortLocation, scaleFactor);
+        }
+    }
+}
 
-            auto endPortLocation = CableViewPortLocationHelper::getPortLocation ({ endEditor, cable->endIdx, true });
+void CableView::resized()
+{
+    for (auto* cable : cables)
+    {
+        cable->setBounds (getLocalBounds());
+    }
+}
 
-            auto startColour = startEditor->getColour();
-            auto endColour = endEditor->getColour();
-            auto levelDB = cable->endProc->getInputLevelDB (cable->endIdx);
-            drawCable (g, startPortLocation.toFloat(), startColour, endPortLocation.toFloat(), endColour, scaleFactor, levelDB);
-        }
-        else if (connectionHelper->cableMouse != nullptr)
+void CableView::mouseDown (const MouseEvent& e)
+{
+    if (e.eventComponent == nullptr)
+        return; // not a valid mouse event
+
+    if (e.mods.isPopupMenu())
+    {
+        for (auto* cable : cables)
         {
-            auto mousePos = connectionHelper->cableMouse->getPosition();
-            const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cable->startProc);
-            if (nearestInputPort.editor != nullptr && nearestInputPort.isInput && ! portLocationHelper->isInputPortConnected (nearestInputPort))
+            juce::Point clicked (e.getMouseDownX(), e.getMouseDownY());
+            if (cable->contains (clicked)) // Checks hitTest method for cable
             {
-                auto endPortLocation = CableViewPortLocationHelper::getPortLocation (nearestInputPort);
-                drawCablePortGlow (g, endPortLocation, scaleFactor);
+                board->popupMenu.showPopupMenu();
```

Comment:
So the issue here is that showing the board popup menu is going to allow the user to replace the cable with any processor, when in this case, the only type of processor that really makes sense is one with a single input and single output.

For example, what happens now if the user tries to replace a cable with a "Tuner" module?

Reply:
Mmm this is a good point. The logic I put in `ProcessorChainActionHelper.cpp` should account for any number of inputs and outputs, I tested it with stereo merger, stereo splitter and tuner and they all work as expected (Or at least how I expect). 

The way I was designing this was I figured it was on the user if they wanted to cut off their signal path with a tuner or something similar.

Let me know if I'm misunderstanding your question or we need to change the logic of how it makes connections/displays potential processors to add.

![Screen Shot 2022-08-10 at 4 15 14 PM](https://user-images.githubusercontent.com/43660096/184029889-13b4bcf3-13d0-4417-97cd-0f4d9cfcd2e1.png)

![Screen Shot 2022-08-10 at 4 15 41 PM](https://user-images.githubusercontent.com/43660096/184029945-63de652b-a89a-44a2-b28e-328ad9329bff.png)

Reply:
Oh interesting... when I was testing the other day, I got a crash when trying to replace the cable with a tuner, and then doing some undo/redo operations. I'm not 100% what the right user experience is here, but if it's working reliably then I don't see a problem with allowing any number of inputs/outputs for now.